### PR TITLE
Saas 12.5 fixes sms bvr

### PR DESCRIPTION
**Pull request by bvr-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sms/wizard/sms_composer.py [hunk 1]
```diff
@@ -183,7 +183,7 @@ def _action_send_sms_mass(self, records=None):
         records = records if records is not None else self._get_records()
 
         sms_record_values = self._prepare_mass_sms_values(records)
-        sms_create_vals = [sms_record_values[record.id] for record in records]
+        sms_create_vals = [sms_record_values[record.id] for record in records if sms_record_values[record.id]['error_code'] != 'sms_number_missing']
```
Comment (tde-banana-odoo) [inline addons/sms/wizard/sms_composer.py]: Please no.

File: addons/sms/static/src/js/message.js [hunk 2]
```diff
@@ -59,7 +59,9 @@ Message.include({
         var sms = _.find(self._smsIds, function (sms) {
             return sms[0] === smsID;
         });
-        sms[2] = smsStatus;
+        if (sms !== undefined && smsStatus !== undefined){
```
Comment (pro-odoo) [inline addons/sms/static/src/js/message.js]: space between ( and {:
`if (sms !== undefined && smsStatus !== undefined) {`

File: addons/mail/models/mail_thread.py [hunk 3]
```diff
@@ -1857,7 +1857,7 @@ def message_post(self,
             'email_from': email_from,
             'model': self._name,
             'res_id': self.id,
-            'body': body,
+            'body': "<br>".join(body.split("\n")),
```
[the remark this answers is not in the document]
Comment (tde-banana-odoo) [reply addons/mail/models/mail_thread.py]: Use plaintext2html instead.

**Comment by pro-odoo** [comment]
@robodoo rebase-ff r+

**Pull request closed without merging**